v3/src/v2/codap-v2-data-context-types.ts

```diff
@@ -205,7 +208,7 @@ export interface ICodapV2GameContextStorage extends ICodapV2DataContextStorage {
 }
 
 export type CodapV2Context = ICodapV2DataContext | ICodapV2GameContext | ICodapV2ExternalContext
-export const isV2ExternalContext = (context: CodapV2Context): context is ICodapV2ExternalContext =>
-  !("type" in context) && ("externalDocumentId" in context)
-export const isV2InternalContext = (context: CodapV2Context): context is ICodapV2DataContext | ICodapV2GameContext =>
-  ("type" in context)
+export const isV2ExternalContext = (context?: CodapV2Context): context is ICodapV2ExternalContext =>
```

Comment:
```suggestion
export const isV2ExternalContext = (context: Maybe<CodapV2Context>): context is ICodapV2ExternalContext =>
```
Using maybe here seems better because it make sure the user passes some argument to function. Calling it without an argument seems like an error.